library/src/lib/menu/menu-item.directive.ts

```diff
@@ -1,41 +1,83 @@
-import { AfterViewInit, Directive, ElementRef, OnChanges } from '@angular/core';
+import { AfterViewInit, Directive, ElementRef, HostListener, OnChanges } from '@angular/core';
+import { Subject } from 'rxjs';
 
 /**
  * The directive that represents a menu item.
  */
 @Directive({
     // TODO to be discussed
     // tslint:disable-next-line:directive-selector
-    selector: '[fd-menu-item]',
+    selector: '[fd-menu-item]'
 })
 export class MenuItemDirective implements AfterViewInit, OnChanges {
+
+    public onKeyDown = new Subject<any>();
+    public onClick = new Subject();
+
+
     /** @hidden */
-    constructor(public itemEl: ElementRef) { }
+    constructor (public itemEl: ElementRef) {}
 
     /** @hidden */
     ngAfterViewInit() {
         if (
             // if the menu item contains a child anchor element, apply 'fd-menu__item' class to that anchor
-            this.itemEl &&
-            this.itemEl.nativeElement &&
-            this.itemEl.nativeElement.children &&
-            this.itemEl.nativeElement.children[0] &&
-            this.itemEl.nativeElement.children[0].tagName === 'A'
+            this.isChildElementAnchor()
         ) {
-            this.itemEl.nativeElement.children[0].classList.add('fd-menu__item');
+            this.getChildrenElements()[0].classList.add('fd-menu__item');
         } else if (
             // if the menu item does not contain a child element, apply 'fd-menu__item' class to the fd-menu-item component
-            this.itemEl &&
-            this.itemEl.nativeElement &&
-            this.itemEl.nativeElement.children &&
-            this.itemEl.nativeElement.children.length === 0
+            this.getChildrenElements() &&
+            this.getChildrenElements().length === 0
         ) {
             this.itemEl.nativeElement.classList.add('fd-menu__item');
         }
     }
 
+    public focus() {
+        if (this.getChildrenElements() && this.getChildrenElements()[0]) {
+            this.getChildrenElements()[0].focus();
+        }
+    }
+
+    @HostListener('keydown', ['$event'])
+    keyDown(event: KeyboardEvent): void {
```

Comment:
Is there a reason why these (+ subject) aren't just events? Technically events are observables too

Reply:
just seems a bit convoluted to have this stuff